ckanext/canada/validators.py

```diff
@@ -491,3 +491,28 @@ def canada_guess_resource_format(key, data, errors, context):
             except ValidationError as e:
                 errors[key].append(e.error_dict['format'])
                 raise StopOnError
+
+
+def protect_registry_access(key, data, errors, context):
+    """
+    Ensure the registry_access field is not changed by an unauthorized user.
+    """
+    if is_sysadmin(context['user']):
+        return
+
+    original = ''
+    org = context.get('group')
+    if org:
+        original = org.extras.get('registry_access', [])
+    value = data.get(key, [])
+
+    if not value:
```

Comment:
Need to check for `missing` here too

Reply:
if you mark it required and add a `scheming_required` validator to the front of the list we won't get missing at this point